automation_tools/__init__.py

```diff
@@ -1655,16 +1656,176 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
-def host_pings(host, timeout=15):
+# =============================================================================
+# Satellite and Capsule Upgrade
+# =============================================================================
+def copy_ssh_key(from_host, to_host):
+    """This will generate(if not already) ssh-key on from_host
+    and copy that ssh-key to to_host.
+
+    Beware that both hosts should have authorized key added
+    for test-running host.
+
+    :param from_host: A string. Hostname on which the key to be generated and
+        to be copied from.
+    :param to_host: A string. Hostname on to which the ssh-key will be copied.
+
+    """
+    execute(lambda: run('[ ! -f ~/.ssh/id_rsa.pub ] && ssh-keygen '
```

Comment:
I don't like this much, but don't see any better way to do this right now. Specially for the scp as fabric provides put and get functions.